coordinator/core/marbleapi.go

```diff
@@ -279,20 +280,34 @@ func customizeParameters(params manifest.Parameters, specialSecrets reservedSecr
 		Secrets:   userSecrets,
 	}
 
+	var err error
+	var newValue string
+
 	// replace placeholders in files
 	for path, data := range params.Files {
-		newValue, err := parseSecrets(data, secretsWrapped)
-		if err != nil {
-			return nil, err
+		if data.NoTemplates {
+			newValue = data.Data
+		} else {
+			newValue, err = parseSecrets(data.Data, secretsWrapped)
+			if err != nil {
+				return nil, err
+			}
 		}
 
 		customParams.Files[path] = []byte(newValue)
 	}
 
 	for name, data := range params.Env {
-		newValue, err := parseSecrets(data, secretsWrapped)
-		if err != nil {
-			return nil, err
+		if data.NoTemplates {
+			newValue = data.Data
+		} else {
+			if strings.Contains(data.Data, string([]byte{0x00})) {
+				return nil, fmt.Errorf("environment variable: %s: content contains null bytes", name)
```

Comment:
Would it not be better if we straight out disallow pumping `raw` to `Env`? It would have failed before too during `Setenv`. While this error message is more user-friendly than before, it is something that can fail occasionally after manifest deployment (e.g. non-shared secrets which are regenerated, user-defined secrets...).

Reply:
`raw` can be used for user-defined secrets, which might contain arbitrary data, including valid utf-8 strings.
Another thing is that most non utf-8 data is still valid data for env variables (except null bytes of course).
So completely disallowing users to use the `raw` keyword for env variables is not a good idea, in my opinion.

This is also not fixed by introducing a new keyword (e.g. utf-8) that only allows valid utf-8 content in env vars, since there might still be a case where a randomly generated key is valid utf-8 and therefore get accepted.
I think it would be best to leave the option, but not advertise it by removing these cases from examples.

Reply:
The reason why I dislike the "support, but not advertise" approach very much in this case is that this allows for spontaneous crashes after deployment. If you define a shared secret, say this:

```json
        "random_crap": {
            "Size": 256,
            "Shared": false,
            "Type": "symmetric-key"
        }
```
and you define this as Env:
```json
                "Env": {
                    "DO_I_BREAK": "{{ raw .Secrets.random_crap }}"
                },
```

And you create new Marbles occasionally... this goes well for quite some time, but some will just crash badly and will repeat to do so.

Having runtime crashes is not a good thing to have, and breaks reliability. It's a game of luck in this case, **and the only way to get rid of this is to redeploy the Coordinator from scratch**, as you cannot update this part of the manifest. Essentially, this allows a Coordinator deployment that's broken and cannot be fixed.

This is very bad UX, in my opinion. A solution such as MarbleRun should always aim for reliability. Sure, we are not in production mode yet, but having crashes by luck _intentionally supported_ is not a good idea.

Of course one can say: "But environment variables can contain more than just UTF-8/Unicode!", well yes... But this would just limit passing raw secrets into the environment. If you manually like to define bytes, you can do so via JSON control characters. It's not like we would completely neuter environment variables, just not support passing potentially random data into it causing crashes by chance.

However, I do not see the use case where someone actually wants to pass bytes through the environment via Secrets. Secrets usually contain cryptographic material, and having cryptographic material with NULL bytes removed _by design_ already seems pretty bad to me. If you use user-defined secrets to pass credentials, well that's usually UTF-8.

I really do not know what the use-case here would be, but supporting this is a reliability issue.

Reply:
I agree this can lead to some annoying situations, but fixing everything that is wrong with templates and env var in their current form goes beyond the scope of this PR.
We should take some time to discuss solutions to this problem and then implement a fix.